src/dayNightSystem/dayNightClock.mjs

```diff
@@ -6,69 +6,101 @@
 // note: don't run more than one day/night clock at a time.  doing so will cause multiple
 //       filters to be applied to the screen, which won't look too nice. :o)
 
-import { Prim, Thread } from 'sphere-runtime';
+import { Thread } from 'sphere-runtime';
 
 import InGameTime from './inGameTime';
 
-const DayMask = Color.Transparent,
-      TwilightMask = new Color(0.5, 0.125, 0.0625, 0.625),
-      NightMask = new Color(0.0, 0.0, 0.125, 0.5625);
+const DayMask = [0.0, 0.0, 0.0, 0.0],
+      TwilightMask = [0.5, 0.125, 0.0625, 0.625],
+      NightMask = [0.0, 0.0, 0.125, 0.5625];
+
+const Day = 0,
+      Night = 1,
+      Fading = 2;
 
 export default
 class DayNightClock extends Thread
 {
 	constructor()
 	{
 		super({ priority: 1 });
+		this.inGameTime = new InGameTime(0, 0, 0);
 
 		console.log("initializing day/night clock", `time: ${this.now()}`);
-		this.currentMask = Color.Transparent;
+		let white = Color.White;
+		let width = Surface.Screen.width;
+		let height = Surface.Screen.height;
+		let rectangle = new Shape(ShapeType.TriStrip,
+			new VertexList([
+				{ x: 0, y: 0, color: white },
+				{ x: width, y: 0, color: white },
+				{ x: 0, y: height, color: white },
+				{ x: width, y: height, color: white },
+			]));
+		this.model = new Model ([rectangle]);
+		this.shader = new Shader({
+			fragmentFile: '#/shaders/image.frag.glsl',
+			vertexFile  : '#/shaders/image.vert.glsl',
+		});
+		this.model.shader = this.shader;
+		this.state = Day;
+		this.currentMask = [0.0, 0.0, 0.0, 0.0];
 		this.start();
 	}
 
 	now()
 	{
-		let realTime = new Date();
-		let currentTime = 3600 * realTime.getHours() + 60 * realTime.getMinutes() + realTime.getSeconds();
-		currentTime = (currentTime * 10) % 86400;
-		let hour = Math.floor(currentTime / 3600);
-		let minute = Math.floor((currentTime / 60) % 60);
-		let second = currentTime % 60;
-		return new InGameTime(hour, minute, second);
+		const time = Date.now();
+		let currentTime = Math.floor(time / 100) % 86400;
```

Comment:
I'm not sure if this is 100% equivalent to what I had before?  To clarify, how this should work is: A Lucidan "week" is 10 in-game days long and always begins at the start of a waking-world day, which is to say: midnight in realtime = midnight of day 1 in-game.  As your version is just "seconds since 1/1/1970, modulo 86400" I think that might put the cycle out of sync?

Reply:
I verified against master, and I was right--the sync is off.  As of this message the version in master reports ~21:00 (night) while yours reports ~13:00 (day)